Replace autostart behavior with hook to start tests.

rwjblue opened the pull request:
Prior to this change, tests automatically started once DOMContentLoaded fires. This means that there is no obvious mechanism to hook into the system after boot, but before 
tests start (e.g. to load lazy engine assets).

This is a somewhat breaking change, apps will have to update their `tests/test-helper.js` to properly kick off the tests.

The change is pretty trivial:

```js
// tests/test-helper.js

import { start } from 'ember-cli-qunit';
// ...snip current content...
start();
```

Closes https://github.com/ember-cli/ember-cli-qunit/issues/171.

trentmwillis commented on addon-test-support/ember-cli-qunit.js:
```diff
@@ -0,0 +1,65 @@
+  }
+});
+
+export class TestLoader extends AbstractTestLoader {
```
Leaving this as a note to myself: if ember-exam breaks, come check this. I don't think it should though.

rwjblue replied:
Gotcha. I just figured this was better than monkey patching the prototype.

trentmwillis replied:
Definitely is, just unsure if it breaks any assumptions in the ember-exam code (been awhile since I looked at it haha)

trentmwillis commented:
Irony: not updating the test-helper.js in this repo 😛

rwjblue commented:
HAHA! Yes, I did forget that, but I also missed some other work that I had deleted and forgot to copy.

Updated now, might want to do another pass...

Turbo87 commented:
@rwjblue do we need something similar for `ember-cli-mocha`?

rwjblue commented:
@Turbo87 - Yes, if you want your API to follow suit...

hjdivad commented:
This has the nice side benefit of improving the ergo for debugging tests that loop, as you can open dev tools before running the tests that cause chrome to thrash.
